src/attach.ts

```diff
@@ -16,23 +16,35 @@ import {
   IndependentContext,
   TracingContext,
 } from "./handler";
+import { SomeNamespaces } from "./namespace";
 import { getRemoteClients } from "./remote-client";
 import { getStartupLogo } from "./startup-logo";
 
-export const attachSockets = async <E extends EmissionMap>({
+interface HookSet<E extends EmissionMap> {
+  /** @desc A place for emitting events regardless receiving events */
+  onConnection?: Handler<ClientContext<E>, void>;
+  onDisconnect?: Handler<ClientContext<E>, void>;
+  onAnyIncoming?: Handler<TracingContext<E>, void>;
+  onAnyOutgoing?: Handler<TracingContext<E>, void>;
+  /** @desc A place for emitting events regardless clients activity */
+  onStartup?: Handler<IndependentContext<E>, void>;
+}
+
+type Hooks<NS extends SomeNamespaces<EmissionMap>> = {
+  [K in keyof NS]?: HookSet<NS[K]>;
+};
+
+export const attachSockets = async <NS extends SomeNamespaces<EmissionMap>>({
   io,
   actions,
```

Comment:
being object, it does not solve the issue of same named action in different namespaces.
requiring `SomeNamespaces` structure here might be redundant.
perhaps it would be better to convert it into an array and allow actions to have names.

Reply:
fixed by 24baa88